### DISCUSSION TITLE
Help creating OSX build with github actions

### DISCUSSION THREAD

**Discussion by EnigmaCurry** [DISTRHO/Cardinal#579] [Q&A] [answered]
For a friend, I'm trying to create an OSX build of my in-development modules (I don't own a mac), so I enabled the github actions on my fork to let github build one for me, but in my test build it skipped several of the steps and it did not produce a binary at the end. I'm not sure why. 

[See the recent test build here](https://github.com/EnigmaCurry/Cardinal/actions/runs/6180233783/job/16776368862)

Here is my [mac-build.yml](https://github.com/EnigmaCurry/Cardinal/blob/mac-build/.github/workflows/mac-build.yml) that I copied the mac section verbatim from the main build.yml (I disabled the main build.yml and irc.yml)

Most likely it is skipping it due to the `if:` conditional for each of these steps:

```
        if: steps.cache.outputs.cache-hit == 'true'
```

but I don't know what would make this evaluate to true (or why it would be false in my case). Any pointers on how to make an OSX build (without owning a mac) are appreciated!

**Comment by falkTX** [comment, marked as answer]
cardinal build takes too long, so it is done in 2 steps for some systems. first it builds all dependencies, and then when those are in cache it builds cardinal.
so just either push a new commit to the repo or manually trigger a rebuild of the last action.

**Reply by EnigmaCurry** [reply]
Oh ok that makes sense I'll trigger a new build then, thanks!
